portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -59,8 +59,25 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
     PreparedQuery results = datastore.prepare(query);
 
+    // Get number of comments to display
+    int numCommentsToDisplay;
+    try {
+      numCommentsToDisplay = Integer.parseInt(request.getParameter("limit"));
+      System.err.println(request.getParameter("limit"));
+    } catch (NumberFormatException e) {
+      System.err.println("Input for limit invalid. Displaying default number.");
+      numCommentsToDisplay = 5;
```

Comment:
Another way to handle a bad input, is to not execute the rest of the servlet and return an HTTP error. Not required for this starter project but something useful that you will probably use a lot in the capstone.

https://tomcat.apache.org/tomcat-5.5-doc/servletapi/javax/servlet/http/HttpServletResponse.html#sendError(int,%20java.lang.String)